common/scala/src/main/scala/org/apache/openwhisk/core/containerpool/logging/DockerToActivationLogStore.scala

```diff
@@ -113,29 +113,61 @@ class DockerToActivationLogStore(system: ActorSystem) extends LogStore {
     logsWithPossibleError
   }
 
+  /**
+   * Determine whether the passed activation log had a log collecting error or not.
+   * It is expected that the log collecting stream appends a message from a well known
+   * set of error messages if log collecting failed.
+   *
+   * If the activation failed due to a developer error, an additional error message is appended.
+   * In that case, the second last message indicates whether there was a log collecting error AND
+   * the last message MUST be the additional error message mentioned above.
+   *
+   * @param actLogs the activation logs to check
+   * @param logLimit the log limit applying to the activation
+   * @param isDeveloperError did activation fail due to developer error?
+   * @return true if log collecting failed, false otherwise
+   */
+  protected def isLogCollectingError(actLogs: ActivationLogs,
```

Comment:
The logic here kinda what I was fearing earlier 😂 . Should we at least leave a TODO to return errors from the log gathering itself and append the appropriate lines based on these after the fact?

Reply:
Well... I feel ashamed about this piece of code... at the same time, I had the feeling that I would have to improve other areas as well.

For example: the log reading pipeline recovers exceptions with regular stream elements, i.e. log messages that we check for later on. If an action just logged one of these messages as last log line, we would detect this a log collecting error...

I tried a lot of different approaches and one of the problems was that the `DockerToActivationFileLogStore` has a fan-out to different sinks so that the "injection" of error messages needs to be performed at a reasonable stage of the pipeline. That probably requires a custom flow or sink.

I guess it's a good idea to leave a TODO in the comments.